src/main/java/com/google/devtools/build/lib/analysis/RuleContext.java

```diff
@@ -409,6 +410,38 @@ public ActionOwner getActionOwner() {
     return getActionOwner(DEFAULT_EXEC_GROUP_NAME);
   }
 
+  /**
+   * Returns a special action owner for test actions. Test actions should run on the target platform
+   * rather than the host platform. Note that the value is not cached (on the assumption that this
+   * method is only called once).
+   */
+  public ActionOwner getTestActionOwner() {
+    PlatformInfo testExecutionPlatform;
+    ImmutableMap<String, String> testExecProperties;
+
+    // If we have a toolchain, pull the target platform out of it.
+    if (toolchainContexts != null) {
+      // TODO(austin.schuh@bluerivertech.com): This doesn't respect execution properties coming from
```

Comment:
File an issue and link that here, rather than a person.

Reply:
Done